core/common/lib/keys-lib/src/main/java/org/eclipse/edc/keys/KeyParserRegistryImpl.java

```diff
@@ -37,4 +38,12 @@ public Result<Key> parse(String encoded) {
                 .map(kp -> kp.parse(encoded))
                 .orElseGet(() -> Result.failure("No parser found that can handle that format."));
     }
+
+    @Override
+    public Result<PublicKey> parsePublic(String encoded) {
+        return parsers.stream().filter(kp -> kp.canHandle(encoded))
+                .findFirst()
+                .map(kp -> kp.parsePublic(encoded).compose(k -> Result.success((PublicKey) k)))
```

Comment:
the type validation that was in `LocalPublicKeyServiceImpl` is missing

Reply:
as per the contract of the `parsePublic` method this can never be something other than a `PublicKey`